app/assets/javascripts/components/article_finder/article_finder_row.jsx

```diff
@@ -1,9 +1,37 @@
 import React from 'react';
 import createReactClass from 'create-react-class';
-
 import { fetchStates } from "../../constants";
 
 const ArticleFinderRow = createReactClass({
+  getInitialState() {
+    return {
+      isAdding: false,
+    };
+  },
+
+  componentWillReceiveProps(nextProps) {
+    if (this.state.isAdding && (this.props.isAdded !== nextProps.isAdded)) {
+      this.setState({
+        isAdding: false,
+      });
+    }
+  },
+
+  addAvailableArticle() {
+    const assignment = {
+      title: decodeURIComponent(this.props.title).trim(),
+      project: this.props.course.home_wiki.project,
+      language: this.props.course.home_wiki.language,
+      course_id: this.props.courseSlug,
+      user_id: null,
+      role: 0,
```

Comment:
Avoid this magic number. We use the assignment role magic numbers 0 (assigned) and 1 (reviewing) through the frontend, but this is as good a time as any to start using named constants for assignment roles instead.